air/src/lib.rs

```diff
@@ -93,14 +127,46 @@ impl Air for ProcessorAir {
         result[0] = next[CLK_COL_IDX] - (current[CLK_COL_IDX] + E::ONE);
 
         // --- range checker ----------------------------------------------------------------------
-        range::enforce_constraints::<E>(frame, &mut result[1..]);
+        RangeCheckerConstraints::new()
+            .enforce_constraints(frame, &mut result[self.range_checker_start()..]);
+
+        // --- auxiliary table of co-processors (hasher, bitwise, memory) -------------------------
+        AuxTableConstraints::new()
+            .enforce_constraints(frame, &mut result[self.aux_table_start()..]);
```

Comment:
It wasn't possible to call enforce_constraints on an instance variable from evaluate_transition because of the generic type. Also, the previous version was instantiating the VM transition every time, so I thought this might be the preferred approach vs having an instance variable.

Once we do away with the function-based generic transition constraint approach this should be doable, and it's my preferred approach as well, given that you don't have objections to adding these structs as instance variables.